rotkehlchen/tests/unit/decoders/test_makerdao_sai.py

```diff
@@ -2299,31 +2300,23 @@ def test_makerdao_sai_proxy_interaction(ethereum_transaction_decoder):
 @pytest.mark.vcr(filter_query_parameters=['apikey'])
 @pytest.mark.parametrize('ethereum_accounts', [['0xca482bCd75A6E0697aD6A1732aa187310b8372Df']])
 def test_makerdao_sai_cdp_migration(ethereum_transaction_decoder, ethereum_accounts):
+    """Data taken from
+    https://etherscan.io/tx/0x03620c6bf5edb7a7935953337ffcfac70d631cf2012d6c80d36828d636063318
+    """
+
     """Check that a Sai CDP migration is decoded properly"""
     tx_hex = deserialize_evm_tx_hash('0x03620c6bf5edb7a7935953337ffcfac70d631cf2012d6c80d36828d636063318 ')  # noqa: E501
     evmhash = deserialize_evm_tx_hash(tx_hex)
-    user_address = ethereum_accounts[0]
+    # TODO: ROTKI TEAM: The address you used here is the DSProxy address, not the user address.
+    # Look at my test_makerdao.py to process both
+    dsproxy_address = ethereum_accounts[0]
     events, _ = get_decoded_events_of_transaction(
         evm_inquirer=ethereum_transaction_decoder.evm_inquirer,
         database=ethereum_transaction_decoder.database,
         tx_hash=evmhash,
     )
     expected_events = [
         EvmEvent(
-            tx_hash=evmhash,
-            sequence_index=0,
-            timestamp=1579044372000,
-            location=Location.ETHEREUM,
-            event_type=HistoryEventType.WITHDRAWAL,
-            event_subtype=HistoryEventSubType.REMOVE_ASSET,
-            asset=A_ETH,
-            balance=Balance(FVal('0.022255814')),
-            location_label='0xca482bCd75A6E0697aD6A1732aa187310b8372Df',
-            notes='Withdraw 0.022255814 ETH from ETH-A MakerDAO vault',
-            counterparty=CPT_VAULT,
-            address=string_to_evm_address('0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2'),
```

Comment:
@LefterisJP , uou can see from this, how the processing of events was bugged. This "0xC02aaA...." is the WETH contract address, and here the event was detected as one "withdraw from ETH-A MakerDAO vault" operation. Also notice the sequence_index number, 0. Basically the decoding here was working "by chance". I think the general decoding could be re-worked for being more deterministic (and reproducible). I have some idea in mind if you are interested on discussing possible sw requirements and architecture.

Reply:
The address field is manually added per decoder and means different things depending on the decoder. Coce can have bugs.

Always open to suggestions, but only things that make sense but for things that make sense you would need a perfect understanding of how the current thing we have works. And that should be in discord, not here.